internals/resource-managers/knative_service.go

```diff
@@ -84,12 +85,12 @@ func (k KnativeServiceManager) CleanUp(capp rcsv1alpha1.Capp) error {
 
 // isRequired determines if a Knative service (ksvc) is required based on the Capp's spec.
 func (k KnativeServiceManager) isRequired(capp rcsv1alpha1.Capp) bool {
-	return !utils.DoesHaltAnnotationExist(capp.Annotations)
+	return capp.Spec.State == cappEnabledState
 }
 
 // isResumed determines if a given Capp resource has resumed its operation.
```

Comment:
Maybe elaborate on the comment here that Resume means that the Capp was `disabled` and has been changed to `enabled`.